resources/component_runit_service.rb

```diff
@@ -0,0 +1,80 @@
+# A runit service wrapper for a chef component
+#
+# :package defines which part of the node attributes we should look in for configuration
+# as it different for each add-on e.g. :package => 'reporting' or :package => 'pushy'
+#
+resource_name :component_runit_service
+
+property :component, String, name_property: true
+property :package, String, default: 'private_chef'
+property :log_directory, [String, nil], default: nil
```

Comment:
Is the intent to allow people to explicitly pass in nils here? If not you can leave out all the nil stuff since any property with a type of String defaults to nil.

Reply:
Mostly intended to keep the defaults from the previous incarnation of the resource definition.

Reply:
nil is still the default even if you don't list it

Reply:
Gotcha. Will simplify.